### PR TITLE
Merging matmathutils and markerutils

### PR DESCRIPTION
**Pull request by jina2k**
Combined MatMathUtils class with MarkerUtils class. Also cleaned up a bit of duplicated code.

Added inheritance so that MarkerOffset extends Pose, and changed private variables to protected so they could be accessed by MarkerOffset.

### CODE REVIEW

File: src/markerdetector/Pose.java [hunk 1]
```diff
@@ -160,14 +160,15 @@ public Pose flipCoords(){
...
-		translationVector3D = MarkerUtils.scalarMultiply(MarkerUtils.matMultiply(r, tvec), -1);
+		translationVector3D = MatMathUtils.scalarMultiply(MatMathUtils.matMultiply(r, tvec), -1);
 		Calib3d.Rodrigues(r, rotationVector3D);
 		return new Pose(rotationVector3D, translationVector3D);
 	}
 
 	/**Returns a hash code for this Pose.
 	@return a hash code for this Pose.
 	*/
+	@Override
```
Comment (Jacob-John8) [inline src/markerdetector/Pose.java:171]: The @Override 'flag' only needs to appear above the sub-class's function and not the super-class's, you should remove these for all the functions in Pose.java

**Review by Jacob-John8 — COMMENTED** [review]
Other than the one comment, the inheritance stuff looks good

File: src/markerdetector/Pose.java (hunk 1 above)
Comment (jina2k) [reply src/markerdetector/Pose.java:171]: Override is necessary for when you override hashCode, toString, and equals because those are object methods

Comment (Jacob-John8) [reply src/markerdetector/Pose.java:171]: you right, you right

**Comment by Jacob-John8** [comment]
I tested the new branch with the MaskSim and TensorSim, both appear to work as intended

**Review by alexcwu1121 — APPROVED** [review]

**Pull request merged**